src/components/legend/draw.js

```diff
@@ -586,23 +588,54 @@ function computeLegendDimensions(gd, groups, traces) {
         extraWidth = 40;
     }
     else if(isGrouped) {
-        var groupXOffsets = [opts._width];
+        var maxHeight = 0;
+        var maxWidth = 0;
         var groupData = groups.data();
 
-        for(var i = 0, n = groupData.length; i < n; i++) {
-            var textWidths = groupData[i].map(function(legendItemArray) {
+        var maxItems = 0;
+
+        groupData.forEach(function(group) {
```

Comment:
**NON-BLOCKING**

We could probably clean up these few lines a little:

- use for-loop here instead of `forEach`
- :zap: that double empty new line
- use [`Lib.aggNums`](https://github.com/plotly/plotly.js/blob/12ce7253e5a3995824ec514a42be01aadbfdb6ea/src/lib/stats.js#L28-L44) instead of `Math.max.apply` and `group.reduce`
- make the `for(var i = 0, n = groupData.length; i < n; i++)` loops `for(var i = 0; i < groupData.length; i++)`, i.e. no need to define `n` (I think).

Reply:
Done in a5fa247 except for `group.reduce`.

Can I really replace `group.reduce` with  `Lib.aggNums`?

Reply:
> Can I really replace `group.reduce` with `Lib.aggNums`?

Maybe with something like:

```js
Lib.aggNums(function(a, b) { return a + b[0].height; }, 0, groups)
```

similar to [`Lib.mean`](https://github.com/plotly/plotly.js/blob/12ce7253e5a3995824ec514a42be01aadbfdb6ea/src/lib/stats.js#L54-L57), but you're right, that's probably an overkill here as all the `b[0].height` should be defined. :dancer: